File: aat-lib/src/main/java/ch/bailu/aat_lib/preferences/SolidStaticIndexList.java
package ch.bailu.aat_lib.preferences;

public class SolidStaticIndexList extends SolidIndexList {

    private final String[] labelList;

    public SolidStaticIndexList(StorageInterface s, String k, String[] l) {
        super(s,k);
        labelList=l;
    }


    @Override
    public int length() {
        return labelList.length;
    }

    @Override
    public String getValueAsString(int i) {
        return labelList[i];
    }


    @Override
    public String[] getStringArray() {
        return labelList;
    }


}

File: aat-lib/src/main/java/ch/bailu/aat_lib/preferences/SolidStaticIndexList.kt
package ch.bailu.aat_lib.preferences

open class SolidStaticIndexList(storage: StorageInterface, key: String, private val labelList: Array<String>) : SolidIndexList(storage, key) {
    override fun length(): Int {
        return labelList.size
    }

    public override fun getValueAsString(index: Int): String {
        return labelList[index]
    }

    override fun getStringArray(): Array<String> {
        return labelList
    }
}
